app/src/main/java/at/bitfire/icsdroid/db/entity/Subscription.kt

```diff
@@ -0,0 +1,396 @@
+package at.bitfire.icsdroid.db.entity
+
+import android.accounts.Account
+import android.content.ContentProviderClient
+import android.content.ContentUris
+import android.content.Context
+import android.database.SQLException
+import android.net.Uri
+import android.os.RemoteException
+import android.provider.CalendarContract
+import android.provider.CalendarContract.Calendars
+import android.provider.CalendarContract.Events
+import android.util.Log
+import androidx.annotation.ColorInt
+import androidx.annotation.WorkerThread
+import androidx.core.content.contentValuesOf
+import androidx.room.Entity
+import androidx.room.Ignore
+import androidx.room.PrimaryKey
+import at.bitfire.ical4android.AndroidCalendar
+import at.bitfire.ical4android.CalendarStorageException
+import at.bitfire.ical4android.util.MiscUtils.UriHelper.asSyncAdapter
+import at.bitfire.icsdroid.Constants.TAG
+import at.bitfire.icsdroid.db.AppDatabase
+import at.bitfire.icsdroid.db.LocalCalendar
+import java.io.FileNotFoundException
+import java.net.MalformedURLException
+
+/**
+ * Represents the storage of a subscription the user has made.
+ * @param id The id of the subscription in the database.
+ * @param url URL of iCalendar file
+ * @param eTag iCalendar ETag at last successful sync
+ * @param displayName Display name of the subscription
+ * @param lastModified iCalendar Last-Modified at last successful sync (or 0 for none)
+ * @param lastSync time of last sync (0 if none)
+ * @param errorMessage error message (HTTP status or exception name) of last sync (or null)
+ * @param ignoreEmbeddedAlerts Setting: whether to ignore alarms embedded in the Webcal
+ * @param defaultAlarmMinutes Setting: Shall a default alarm be added to every event in the calendar? If yes, this field contains the minutes before the event.
+ * If no, it is `null`.
+ * @param color The color that represents the subscription.
+ */
+@Entity(tableName = "subscriptions")
+data class Subscription(
+    @PrimaryKey val id: Long = 0L,
+    val url: Uri,
+    val eTag: String? = null,
+
+    val displayName: String,
+
+    val accountName: String,
+    val accountType: String,
```

Comment:
I have been revising this, and now I'm thinking that since the name of the account is localized, if the user changes the language of the phone the subscriptions will be unlinked

Reply:
73577da793adf2f32a0bbeac5745165f25d94c66